### PR TITLE
update to java 8 and drop joda-time

### PR DESCRIPTION
**Pull request by codylerum**
For #131

### CODE REVIEW

File: src/main/java/com/lob/exception/APIException.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 public class APIException extends LobException {
 
+    private static final long serialVersionUID = 1L;
```
Comment (hwrdprkns) [inline src/main/java/com/lob/exception/APIException.java]: Is there any detriment in letting the compiler generate these. I'm curious if they are being used in implementations somewhere

Comment (codylerum) [reply src/main/java/com/lob/exception/APIException.java]: IDE dropped them in. 

There shouldn't be any issue with them being there but let me rip them out for consistency.

File: src/main/java/com/lob/exception/InvalidRequestException.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import java.util.Map;
 
 public class InvalidRequestException extends LobException {
-
+
```
Comment (hwrdprkns) [inline src/main/java/com/lob/exception/InvalidRequestException.java]: Spacing clean up

**Comment by codylerum** [comment]
@hwrdprkns Actually looks like LobException already had it so just adding to others to keep it consistent.

I also updated RequestOptions to give it a cleaner equals and a hashCode so you might want to give it another review pass.

**Pull request merged**